packages/sdks/snippets/react/src/routes/custom-components/advanced-child.tsx

```diff
@@ -0,0 +1,92 @@
+
+
+/**
+ * Quickstart snippet
+ * snippets/react/routes/custom-child.tsx
+ */
+
+import {
+    Content,
+    fetchOneEntry,
+    getBuilderSearchParams,
+    isPreviewing,
+    type BuilderContent,
+  } from '@builder.io/sdk-react';
+  import { useEffect, useState } from 'react';
+import CustomColumns from '../../components/CustomColumns';
+  
+  const BUILDER_API_KEY = 'ee9f13b4981e489a9a1209887695ef2b';
+  const MODEL_NAME = 'page';
+  
+  // set whether you're using the Visual Editor,
+  // whether there are changes,
+  // and render the content if found
+  export default function AdvancedChildRoute() {
+    const [notFound, setNotFound] = useState(false);
+    const [content, setContent] = useState<BuilderContent | null>(null);
+  
+    // get the page content from Builder
+    useEffect(() => {
+      fetchOneEntry({
+        model: MODEL_NAME,
+        apiKey: BUILDER_API_KEY,
+        userAttributes: {
+          urlPath: window.location.pathname,
+        },
+        options: getBuilderSearchParams(new URL(location.href).searchParams),
+      })
+        .then((content) => {
+          if (content) {
+            setContent(content);
+          }
+          setNotFound(!content);
+        })
+        .catch((err) => {
+          console.log('Oops: ', err);
+        });
+    }, []);
+  
+    // If no page is found, return
+    // a 404 page from your code.
+    if (notFound && !isPreviewing()) {
+      return <div>404</div>;
+    }
+  
+    // return the page when found
+    return (
+      <Content
+        content={content}
+        model={MODEL_NAME}
+        apiKey={BUILDER_API_KEY}
+        customComponents={[
+            {
+                name: "MyColumns",
+                component: CustomColumns,
+                shouldReceiveBuilderProps: {
+                  builderBlock: true,
+                },
+                inputs: [
+                  {
+                    name: "inside",
+                    type: "array",
+                    defaultValue: [
+                      {
+                        blocks: [],
+                      },
+                      {
+                        blocks: [],
+                      },
+                    ],
+                    subFields: [{
+                      name: "blocks",
+                      type: 'uiBlocks'
+                    }],
+                  },
+                ],
+              },
+   
```

Comment:
let's move this to a separate object called `customComponents`. We can also import and use the `RegisteredComponent` type from the SDK for it:

```ts
import { RegisteredComponent } from '@builder.io/sdk-react';

const customComponents: RegisteredComponent[] = [{
  //...
}]
```

Reply:
What if we have 3 custom components (CustomHero, CustomTabs, CustomAdvancedChild)? Should I have to use the same customComponents function, or do I need to create 3 different files?

Reply:
Update: 
I have added all objects into customComponents as suggested, and then I created one utility function to extract the required obj for a corresponding custom component like this: 

```import { RegisteredComponent } from '@builder.io/sdk-react';

export function extractRelevantComponentByName(name: string) {
  return function (arr: RegisteredComponent[]) {
    const filteredComponents = arr.filter(
      (component) => component.name === name
    );
    return filteredComponents.length > 0 ? filteredComponents : [];
  };
}
```

Please let me know if this is not what I'm supposed to do.